File: Aces/Trainer.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Aces
{
    public class Trainer : Pilot
    {
        string[] movePattern;
        int index;

        public Trainer(string[] movePattern)
        {
            this.movePattern = movePattern;
            if (this.movePattern == null || this.movePattern.Length == 0)
            {
                this.movePattern = new string[] { "Straight" };
            }
            this.index = 0;
        }
        override public string getNextManeuver(int page, string playerMove)
        {
            string result = this.movePattern[this.index];
            this.index = (this.index + 1) % movePattern.Length;
            return result;
        }

        override public int HitPoints { get { return 8; } }

        public override int GetDamage(int range)
        {
            return 0;
        }
    }
}

File: ace-of-aces-react/src/models/trainer.ts
// src/models/trainer.ts

import { Pilot } from './player';

export class Trainer extends Pilot {
  private movePattern: string[];
  private index: number;

  constructor(movePattern: string[]) {
    super();
    this.movePattern = movePattern;
    if (!this.movePattern || this.movePattern.length === 0) {
      this.movePattern = ["Straight"];
    }
    this.index = 0;
  }

  getNextManeuver(_page: number, _playerMove: string): string {
    const result = this.movePattern[this.index];
    this.index = (this.index + 1) % this.movePattern.length;
    return result;
  }

  getDamage(_range: number): number {
    return 0;
  }
}